tidal-link/src/hs/Sound/Tidal/Clock.hs

```diff
@@ -288,42 +295,10 @@ getCycleTime :: ClockConfig -> ClockRef -> IO Time
 getCycleTime config (ClockRef _ abletonLink) = do
                             now <- Link.clock abletonLink
                             ss <- Link.createAndCaptureAppSessionState abletonLink
-                            c <- timeToCycles' config ss now
+                            c <- timeToCycles config ss now
                             Link.destroySessionState ss
                             return $! c
 
--- onSingleTick assumes it runs at beat 0.
--- The best way to achieve that is to use forceBeatAtTime.
--- But using forceBeatAtTime means we can not commit its session state.
--- Another session state, which we will commit,
--- is introduced to keep track of tempo changes.
-getZeroedLinkOperations :: ClockConfig -> ClockRef -> IO LinkOperations
-getZeroedLinkOperations config (ClockRef _ abletonLink) = do
-              sessionState <- Link.createAndCaptureAppSessionState abletonLink
-              zeroedSessionState <- Link.createAndCaptureAppSessionState abletonLink
-
-              nowOsc <- O.time
-              nowLink <- Link.clock abletonLink
-
-              Link.forceBeatAtTime zeroedSessionState 0 (nowLink + processAhead) (cQuantum config)
-
-              Link.commitAndDestroyAppSessionState abletonLink sessionState
```

Comment:
Could this be the source of the bug? We are running commitAndDestroy immediately - before the IO where the session states are used. Compare to clockProcess where I understand we first process the arc using liftIO, then we run commitAndDestroy.

Reply:
The suggestion in https://github.com/tidalcycles/Tidal/pull/1090/files/81c19870a81294e4b5def9cdc24365fa0b23e726#r1645196629 could be applied here. The callback would invoke both destroy functions.

Reply:
hmm yes that makes sense, the callback is probably a good idea! thanks!